pkg/liquidity-source/integral/pool_simulator.go

```diff
@@ -0,0 +1,202 @@
+package integral
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/goccy/go-json"
+	"github.com/holiman/uint256"
+
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/entity"
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/source/pool"
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/util/bignumber"
+	"github.com/KyberNetwork/logger"
+)
+
+func NewPoolSimulator(entityPool entity.Pool) (*PoolSimulator, error) {
+	var pair IntegralPair
+	if err := json.Unmarshal([]byte(entityPool.Extra), &pair); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal Extra: %v", err)
+	}
+
+	numTokens := len(entityPool.Tokens)
+	tokens := make([]string, numTokens)
+	reserves := make([]*big.Int, numTokens)
+
+	for i := 0; i < numTokens; i++ {
+		tokenAddr := entityPool.Tokens[i].Address
+		tokens[i] = tokenAddr
+		reserves[i] = bignumber.NewBig(entityPool.Reserves[i])
+	}
+
+	return &PoolSimulator{
+		Pool: pool.Pool{
+			Info: pool.PoolInfo{
+				Address:    entityPool.Address,
+				ReserveUsd: entityPool.ReserveUsd,
+				Exchange:   entityPool.Exchange,
+				Type:       entityPool.Type,
+				Tokens:     tokens,
+				Reserves:   reserves,
+				Checked:    false,
+			},
+		},
+		IntegralPair: IntegralPair{
+			RelayerAddress: pair.RelayerAddress,
+			IsEnabled:      pair.IsEnabled,
+			SwapFee:        pair.SwapFee,
+			X_Decimals:     pair.X_Decimals,
+			Y_Decimals:     pair.Y_Decimals,
+			AveragePrice:   pair.AveragePrice,
+			SpotPrice:      pair.SpotPrice,
+			Token0LimitMin: pair.Token0LimitMin,
+			Token1LimitMin: pair.Token1LimitMin,
+		},
```

Comment:
can we directly use `pair` here?
```suggestion
		IntegralPair: pair,
```